cmd/clusterctl/cmd/plugin_test.go

```diff
@@ -0,0 +1,103 @@
+/*
+Copyright 2025 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package cmd
+
+import (
+	"bytes"
+	"fmt"
+	"os"
+	"os/exec"
+	"path/filepath"
+	"testing"
+
+	. "github.com/onsi/gomega"
+)
+
+var baseArgs = []string{"clusterctl", "foo"}
+
+const (
+	forkEnvVar          = "CLUSTERCTL_PLUGIN_TEST_FORK"
+	pluginCommandEnvVar = "CLUSTERCTL_PLUGIN_COMMAND"
+)
+
+func Test_plugin(t *testing.T) {
+	if os.Getenv(forkEnvVar) == "" {
+		g := NewWithT(t)
+		tt := []struct {
+			name     string
+			command  string
+			expected string
+		}{
+			{
+				name:     "base plugin command test",
+				expected: "I am a plugin named clusterctl-foo",
+			},
+			{
+				name:     "plugin version command test",
+				command:  "version",
+				expected: "1.0.0",
+			},
+		}
+
+		tmpDir := os.TempDir()
+		defer os.Remove(tmpDir)
+
+		pluginPath := filepath.Join(tmpDir, "clusterctl-foo")
+		path := os.Getenv("PATH")
+		g.Expect(os.WriteFile(pluginPath, []byte(pluginCode), 0755)).To(Succeed()) //nolint:gosec
+
+		for _, tc := range tt {
+			t.Run(tc.name, func(t *testing.T) {
+				g := NewWithT(t)
+				stdout, _, err := runForkTest(t.Name(), fmt.Sprintf("%s=%s", pluginCommandEnvVar, tc.command), fmt.Sprintf("PATH=%s:%s", tmpDir, path))
+				g.Expect(err).NotTo(HaveOccurred())
+				g.Expect(stdout).To(ContainSubstring(tc.expected))
```

Comment:
Let's re-arrange this a bit :-) just cosmetics and adding some godocs.

```suggestion
		g := NewWithT(t)
		tmpDir := os.TempDir()
		defer os.Remove(tmpDir)

		// Create a bash based plugin used for the test which gets added to the PATH variable.
		pluginPath := filepath.Join(tmpDir, "clusterctl-foo")
		pathVar := os.Getenv("PATH")
		g.Expect(os.WriteFile(pluginPath, []byte(pluginCode), 0755)).To(Succeed()) //nolint:gosec

		tt := []struct {
			name     string
			command  string
			expected string
		}{
			{
				name:     "base plugin command test",
				expected: "I am a plugin named clusterctl-foo",
			},
			{
				name:     "plugin version command test",
				command:  "version",
				expected: "1.0.0",
			},
		}
		for _, tc := range tt {
			t.Run(tc.name, func(t *testing.T) {
				g := NewWithT(t)
				stdout, _, err := runForkTest(t.Name(), fmt.Sprintf("%s=%s", pluginCommandEnvVar, tc.command), fmt.Sprintf("PATH=%s:%s", tmpDir, pathVar))
				g.Expect(err).To(Succeed())
				g.Expect(stdout).To(ContainSubstring(tc.expected))
```